### PR TITLE
Public interfaces FullTextSearch

### PR DESCRIPTION
**Pull request by ArtificialOwl**
Signed-off-by: Maxence Lange <maxence@artificial-owl.com>

### CODE REVIEW

File: lib/private/FullTextSearch/FullTextSearchManager.php [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	 * @return IIndex
+	 * @throws FullTextSearchAppNotAvailableException
+	 */
+	public function getIndex(string $providerId, string $documentId): IIndex {
+		return $this->getIndexService()->getIndex($providerId, $documentId);
+	}
+
+	/**
+	 * @param string $providerId
+	 * @param string $documentId
```
Comment (ChristophWurst) [inline lib/private/FullTextSearch/FullTextSearchManager.php]: is that a path?

File: lib/private/FullTextSearch/FullTextSearchManager.php [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+	 * @param string $ownerId
```
Comment (ChristophWurst) [inline lib/private/FullTextSearch/FullTextSearchManager.php]: is that a UID? If so, I'd suggest to rename

File: lib/private/FullTextSearch/FullTextSearchManager.php [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+	 * @param int $status
```
Comment (ChristophWurst) [inline lib/private/FullTextSearch/FullTextSearchManager.php]: are the allowed values documented anywhere? could you add these as interface constants?

File: lib/public/FullTextSearch/Model/IIndex.php [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+	 * @since 15.0.0
+	 *
+	 * @param string $ownerId
+	 *
+	 * @return IIndex
+	 */
+	public function setOwnerId(string $ownerId): IIndex;
+
+	/**
+	 * Get the owner of the originl document.
```
Comment (ChristophWurst) [inline lib/public/FullTextSearch/Model/IIndex.php]: typo

**Review by ChristophWurst — COMMENTED** [review]
Structure looks good :heavy_check_mark: 
Private API is minimal :heavy_check_mark: 
Public classes are final :heavy_check_mark: 

Can't review the semantics of the APIs because I lack the full text search knowledge.

Found a few remarks in the code: naming and a typo.

File: lib/private/FullTextSearch/FullTextSearchManager.php (hunk 1 above)
Comment (daita) [reply lib/private/FullTextSearch/FullTextSearchManager.php]: no its an Id, could be a fileId, a bookmarkId, any Id

File: lib/private/FullTextSearch/FullTextSearchManager.php (hunk 2 above)
Comment (daita) [reply lib/private/FullTextSearch/FullTextSearchManager.php]: I could rename it to userId

Comment (daita) [reply lib/private/FullTextSearch/FullTextSearchManager.php]: renamed

File: lib/private/FullTextSearch/FullTextSearchManager.php (hunk 3 above)
Comment (daita) [reply lib/private/FullTextSearch/FullTextSearchManager.php]: added a link to the interface

File: lib/public/FullTextSearch/IFullTextSearchManager.php [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+	 * true will reset the status to the value defined in the parameter.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param string $providerId
+	 * @param string $documentId
+	 * @param int $status
+	 * @param bool $reset
+	 *
+	 * @return mixed
```
Comment (rullzer) [inline lib/public/FullTextSearch/IFullTextSearchManager.php]: Is this true?

File: lib/public/FullTextSearch/IFullTextSearchManager.php [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+	 * true will reset the status to the value defined in the parameter.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param string $providerId
+	 * @param array $documentIds
+	 * @param int $status
+	 * @param bool $reset
+	 *
+	 * @return mixed
```
Comment (rullzer) [inline lib/public/FullTextSearch/IFullTextSearchManager.php]: Same here

**Review by rullzer — APPROVED** [review]
Looks good to me. :rocket:

File: lib/public/FullTextSearch/IFullTextSearchManager.php (hunk 6 above)
Comment (daita) [reply lib/public/FullTextSearch/IFullTextSearchManager.php]: wow, thanks. does not returns anything.

File: lib/public/FullTextSearch/IFullTextSearchManager.php (hunk 5 above)
Comment (daita) [reply lib/public/FullTextSearch/IFullTextSearchManager.php]: wow, thanks. does not returns anything.

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**